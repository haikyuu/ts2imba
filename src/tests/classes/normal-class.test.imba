import {build} from "../../index"
import * as imbac from 'imba/compiler'

let tsx-code = '''
class Comp{
	render(){
		return <img />
	}	
}
'''

test 'normal class with jsx' do
	const result = await build tsx-code
	expect(result.code).toMatchSnapshot()
	const out = imbac.compile(result.code, sourceId: 'sth')