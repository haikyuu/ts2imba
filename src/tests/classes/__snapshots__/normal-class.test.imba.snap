// Vitest Snapshot v1

exports[`normal class with jsx 1`] = `
"tag Comp
	<self>
		<img>
"
`;
